crates/sequencer/src/sequencer.rs

```diff
@@ -72,18 +72,21 @@ where
         transaction: Transaction,
     ) -> TransactionExecutionResult<TransactionExecutionInfo> {
         let sender_address = match &transaction {
-            Transaction::AccountTransaction(tx) => match tx {
-                blockifier::transaction::account_transaction::AccountTransaction::Invoke(tx) => {
-                    tx.tx.sender_address()
+            Transaction::Account(account_tx) => {
+                let tx = &account_tx.tx;
+                match tx {
+                    starknet_api::executable_transaction::AccountTransaction::Invoke(tx) => {
+                        tx.sender_address()
+                    }
+                    starknet_api::executable_transaction::AccountTransaction::Declare(tx) => {
+                        tx.sender_address()
+                    }
+                    starknet_api::executable_transaction::AccountTransaction::DeployAccount(tx) => {
+                        tx.contract_address()
+                    }
```

Comment:
Maybe here it will be more readable to directly import `starknet_api::executable_transaction::AccountTransaction` and use `AccountTransaction::Invoke` or `AccountTransaction::Declare` for example.